lib/hash_sockaddr.c

```diff
@@ -96,11 +96,11 @@ void *mptcpd_hash_sockaddr_key_copy(void const *p)
         if (src->sa->sa_family == AF_INET) {
                 sa = (struct sockaddr *) l_new(struct sockaddr_in, 1);
 
-                memcpy(sa, src->sa, sizeof(struct sockaddr_in));
+                memcpy(sa, src->sa, sizeof(*src->sa));
         } else {
                 sa = (struct sockaddr *) l_new(struct sockaddr_in6, 1);
 
-                memcpy(sa, src->sa, sizeof(struct sockaddr_in6));
+                memcpy(sa, src->sa, sizeof(*src->sa));
```

Comment:
This won't work for IPv6 addresses since `struct sockaddr` is not large enough to contain a `struct sockaddr_in6`. For example,  `struct sockaddr` is 16 bytes and `struct sockaddr_in6` is 28 bytes on my platform.  Only the first 16 bytes of the `struct sockaddr_in6` would be copied in this case.